### PR TITLE
Court Order and POA

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#6413

*Description of changes:*
- Implements Court Order number and Plan of arrangement component. 
- Applied to storybook.
- Unit tests.
- Storybook build.

### CODE REVIEW

File: src/components/CourtOrderPoa/CourtOrderPoa.stories.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+}
+
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
+  components: { CourtOrderPoa },
+  template: '<court-order-poa v-bind="$props" />' // $props comes from args below
+})
+
+// const ContactData: ContactPointIF = {
```
Comment (severinbeauvais) [inline src/components/CourtOrderPoa/CourtOrderPoa.stories.ts:16]: Obsolete code to delete?

Also below.

File: src/components/CourtOrderPoa/CourtOrderPoa.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        (v: string) => !(v.length > 20) || 'Court order number is invalid'
+      ]
+      this.$refs.courtNumRef.validate()
+    } else this.$refs.courtNumRef.resetValidation()
+  }
+
+  /** Emit court order number. */
+  @Watch('courtOrderNumber')
+  @Emit('emitCourtNumber')
+  private emitCourtNumber (): string { return this.courtOrderNumber }
```
Comment (severinbeauvais) [inline src/components/CourtOrderPoa/CourtOrderPoa.vue:71]: Oh that's clever (both a watcher and an emitter). 👍

File: src/components/CourtOrderPoa/CourtOrderPoa.stories.ts (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/CourtOrderPoa/CourtOrderPoa.stories.ts:16]: OOPS, lots of clean up required here.

Comment (cameron-freshworks) [reply src/components/CourtOrderPoa/CourtOrderPoa.stories.ts:16]: The story was much simpler than i expected lol

File: src/components/CourtOrderPoa/package.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@bcrs-shared-components/court-order-poa",
+  "version": "0.0.0",
+  "publishConfig": {
+    "access": "public"
+  },
+  "dependencies": {
+    "@bcrs-shared-components/interfaces": "^1.0.14",
```
Comment (severinbeauvais) [inline src/components/CourtOrderPoa/package.json:8]: Verify that this gets updated to 1.0.15 after you run the Lerna commands.

File: src/components/CourtOrderPoa/CourtOrderPoa.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/CourtOrderPoa/CourtOrderPoa.vue:71]: Seems to be a good way to keep the parent in sync with a local model. Useful for shared components, although i'm not sure what the other options are yet! Maybe there is a better way still??

File: src/components/CourtOrderPoa/package.json (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/CourtOrderPoa/package.json:8]: k

File: tests/unit/CourtOrderPoa.spec.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    const wrapper: Wrapper<CourtOrderPoa> = createComponent()
+
+    // Verify checkbox is NOT selected
+    expect(wrapper.vm.$data.planOfArrangement).toBe(false)
+
+    // Prompt validates through prop
+    wrapper.setProps({ validate: true })
+    await Vue.nextTick()
+
+    expect(wrapper.find('#court-num-form').text()).toBe('Court Order Number')
```
Comment (severinbeauvais) [inline tests/unit/CourtOrderPoa.spec.ts:70]: Can you also check the emitted event(s) here (and in unit tests below) please?

Comment (cameron-freshworks) [reply tests/unit/CourtOrderPoa.spec.ts:70]: For sure. I did in the last test when verifying the `valid` state.

File: src/components/CourtOrderPoa/CourtOrderPoa.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/CourtOrderPoa/CourtOrderPoa.vue:71]: Not sure about other options, but for sure having a concise API between parent and component (ie, props + events) is good.

**Review by severinbeauvais — APPROVED** [review]
Just some small comments. Update and merge when ready.

**Comment by severinbeauvais** [comment]
@cameron-freshworks Did this get merged correctly?

**Comment by cameron-freshworks** [comment]
It didn't get merged to master, it was closed after a disastrous rebase. 
There was a lot of conflicts in the build files `docs` etc, which i didn't want to resolve as that would be insane.
I thought i could skip only those but it wiped all my work and set it square with master. 

I recovered the commit from the logs and am just putting up the new PR now 😅

**Pull request closed without merging**